framework/src/executioners/Transient.C

```diff
@@ -82,6 +82,7 @@ InputParameters validParams<Transient>()
   params.addParamNamesToGroup("picard_max_its picard_rel_tol picard_abs_tol", "Picard");
 
   params.addParam<bool>("verbose", false, "Print detailed diagnostics on timestep calculation");
+  params.addParam<unsigned int>("max_xfem_update", std::numeric_limits<unsigned int>::max(), "Maximum number of times to update XFEM crack topology in a step due to evolving cracks");
```

Comment:
Is this a sensible default? I'm not sure how often you are overriding this value but "unsigned int max" for practical purposes is unlimited.